docs: update readme, docs layout and add user, dev docs [skip ci] (#87)
red-hat-storage/lvm-operator: docs/dev-guide/topolvm-csi.md

# Topolvm CSI

- LVM Operator orchestrates Topolvm CSI by creating and managing all required
  resources for seamless usage of Logical Volumes as Persistent Volumes

> - This document outlines only the changes made to deployments of Topolvm CSI
>   components and further info please consult topolvm [docs][topolvm-docs]

Reviewer: ```suggestion
- Please refer to topolvm [docs][topolvm-docs] for more details on topolvm
```

## CSI Driver

- *csiDriver* reconcile unit deploys Topolvm CSIDriver resource

> - Ephemeral volumes support is removed (not in the code) due to below reasons:

Reviewer: The LVM operator does not expose support for ephemeral volumes.

Please remove the section on reasons.

  1. There are chances that topolvm project will support Generic Ephemeral
     Volumes and deprecate CSI ephemeral volume support, when that happens
     LVMO will try to include that support
  2. Technical difficulty on deciding default device class when there are
     multiple volume groups across nodes

## Topolvm Controller

- *topolvmController* reconcile unit deploys a single Topolvm Controller plugin
  deployment and manages any updates to the deployment

> - Topolvm scheduler is not used for pod scheduling but only CSI StorageCapacity
>   tracking feature is used

Reviewer: ```suggestion
- Topolvm scheduler is not used for pod scheduling. The CSI StorageCapacity
  feature is used by the scheduler.
```

- An init container generates openssl certs to be used in topolvm controller
  which will be soon replaced with cert-manager

## Topolvm Node and LVMd

- *topolvmNode* reconcile unit deploys and manages topolvm node plugin and lvmd

>   daemonset and scales it based on node selector on which volume groups exists

Reviewer: ```suggestion
  daemonset based on the node selector specified in the deviceClasses in LVMCluster.
```

> - An init container polls for the availability of lvmd config file before start
>   of lvmd and topolvm-node containers
> 

Reviewer: ```suggestion
- An init container polls for the availability of lvmd config file before starting
  the lvmd and topolvm-node containers

```
